manifests/init.pp

```diff
@@ -117,11 +127,106 @@
     home   => $caddy_home,
   }
 
-  contain caddy::install
-  contain caddy::config
-  contain caddy::service
+  archive { $caddy_dl_dir:
+    ensure       => present,
+    extract      => true,
+    extract_path => $install_path,
+    source       => $caddy_dl_url,
+    username     => $caddy_account_id,
+    password     => $caddy_api_key,
+    user         => 'root',
+    group        => 'root',
+    creates      => "${install_path}/caddy",
+    cleanup      => true,
+    notify       => File_capability["${install_path}/caddy"],
+  }
+
+  file_capability { "${install_path}/caddy":
+    ensure     => present,
+    capability => 'cap_net_bind_service=ep',
+    require    => Archive[$caddy_dl_dir],
+  }
+
+  file { $caddy_home:
+    ensure  => directory,
+    owner   => $caddy_user,
+    group   => $caddy_group,
+    mode    => '0755',
+    require => Archive[$caddy_dl_dir],
+    notify  => Service['caddy'],
+  }
+
+  file { $caddy_ssl_dir:
+    ensure  => directory,
+    owner   => $caddy_user,
+    group   => $caddy_group,
+    mode    => '0755',
+    require => Archive[$caddy_dl_dir],
+    notify  => Service['caddy'],
+  }
+
+  file { $caddy_log_dir:
+    ensure  => directory,
+    owner   => $caddy_user,
+    group   => $caddy_group,
+    mode    => '0755',
+    require => Archive[$caddy_dl_dir],
+    notify  => Service['caddy'],
+  }
+
+  file { '/etc/caddy':
+    ensure  => directory,
+    owner   => 'root',
+    group   => 'root',
+    mode    => '0755',
+    require => Archive[$caddy_dl_dir],
+    notify  => Service['caddy'],
+  }
+
+  file { '/etc/caddy/Caddyfile':
+    ensure  => file,
+    owner   => $caddy_user,
+    group   => $caddy_group,
+    mode    => '0444',
+    source  => 'puppet:///modules/caddy/etc/caddy/Caddyfile',
```

Comment:
I have become a fan of
```suggestion
    content => file('caddy/etc/caddy/Caddyfile'),
```

Reply:
me too, just trying to focus on changing the design pattern with the sub classes. I'll go ahead and change this.

Reply:
Realized that the file system layout was used under files/ and templates/.. Going to clean that up, too. Different platforms place things in different places, so this pattern should not be used.

Reply:
Agreed, but it does blow up the size of the PR. Perhaps that should be its own so it's easier to review?

Reply:
It's pretty simple and straight forward since there are only two files in each of files/ and templates/ that were moved.